### ISSUE TITLE
Runs paru -Pw on --dry-run

### ISSUE THREAD

**Issue by Samisafool** [r-darwish/topgrade#865]
## What did you expect to happen?
Topgrade wouldn't run ANY commands when using the `--dry-run` flag.

## What actually happened?
It ran `paru -Pw` (checking for news) and then dry-ran the rest of the stuff

## Additional details
- Which operating system or Linux distribution are you using?
Arch ~~*btw*~~

- How did you install Topgrade?
AUR. Package name is `topgrade`

<details><pre>
 2022-02-28T14:09:48.655Z DEBUG topgrade::config > Configuration at /home/arch/.config/topgrade.toml
 2022-02-28T14:09:48.655Z DEBUG topgrade::config > Loaded configuration: ConfigFile { pre_commands: Some({}), post_commands: None, commands: Some({"Pacman files DB": "sudo pacman -Fy"}), git_repos: None, predefined_git_repos: None, disable: Some([Yadm]), ignore_failures: None, remote_topgrades: None, remote_topgrade_path: None, ssh_arguments: None, git_arguments: None, tmux_arguments: None, set_title: None, assume_yes: None, yay_arguments: None, no_retry: None, run_in_tmux: None, cleanup: None, notify_each_step: None, accept_all_windows_updates: None, bashit_branch: None, only: None, composer: Some(Composer { self_update: None }), brew: Some(Brew { greedy_cask: None }), linux: Some(Linux { yay_arguments: None, arch_package_manager: None, show_arch_news: None, trizen_arguments: None, dnf_arguments: None, apt_arguments: None, enable_tlmgr: None, redhat_distro_sync: None, rpm_ostree: None, emerge_sync_flags: None, emerge_update_flags: None }), git: Some(Git { max_concurrency: None, arguments: None, repos: None, pull_predefined: None }), windows: Some(Windows { accept_all_updates: None, self_rename: None, open_remotes_in_new_terminal: None }), npm: Some(NPM { use_sudo: None }), vim: None, firmware: Some(Firmware { upgrade: None }), vagrant: None, flatpak: Some(Flatpak { use_sudo: None }) }
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Detected "/usr/bin/notify-send" as "notify-send"
 2022-02-28T14:09:48.655Z DEBUG topgrade         > Version: 8.2.0
 2022-02-28T14:09:48.655Z DEBUG topgrade         > OS: x86_64-unknown-linux-gnu
 2022-02-28T14:09:48.655Z DEBUG topgrade         > Args { inner: ["topgrade", "-v"] }
 2022-02-28T14:09:48.655Z DEBUG topgrade         > Binary path: Ok("/usr/bin/topgrade")
 2022-02-28T14:09:48.655Z DEBUG topgrade         > Self Update: false
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Detected "/usr/bin/git" as "git"
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Cannot find "doas"
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Detected "/usr/bin/sudo" as "sudo"
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Cannot find "pwsh"
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Cannot find "powershell"
 2022-02-28T14:09:48.655Z DEBUG topgrade::runner > Step "System update"

―― 19:39:48 - System update ――――――――――――――――――――――――――――――――――――――――――――――――――――
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Cannot find "powerpill"
 2022-02-28T14:09:48.655Z DEBUG topgrade::utils  > Detected "/usr/bin/paru" as "paru"
no new news
 2022-02-28T14:09:50.423Z DEBUG topgrade::executor > Running "/usr/bin/paru" "--pacman" "pacman" "-Syu" "--devel"
:: Synchronizing package databases...
 core-x86-64-v3 is up to date
 extra-x86-64-v3 is up to date
 community-x86-64-v3 is up to date
 core is up to date
 extra is up to date
 community is up to date
:: Starting full system upgrade...
 there is nothing to do
:: Looking for AUR upgrades
:: Looking for devel upgrades
 there is nothing to do
 2022-02-28T14:09:55.383Z DEBUG topgrade::runner   > Step "config-update"
 2022-02-28T14:09:55.383Z DEBUG topgrade::utils    > Detected "/usr/bin/pacdiff" as "pacdiff"
 2022-02-28T14:09:55.383Z DEBUG topgrade::utils    > Detected "/usr/bin/vim" as "vim"

―― 19:39:55 - Configuration update ―――――――――――――――――――――――――――――――――――――――――――――
 2022-02-28T14:09:55.383Z DEBUG topgrade::executor > Running "/usr/bin/sudo" "/usr/bin/pacdiff"
 2022-02-28T14:09:55.607Z DEBUG topgrade::runner   > Step "Brew"
 2022-02-28T14:09:55.607Z DEBUG topgrade::runner   > Step "nix"
 2022-02-28T14:09:55.607Z DEBUG topgrade::runner   > Step "home-manager"
 2022-02-28T14:09:55.607Z DEBUG topgrade::runner   > Step "asdf"
 2022-02-28T14:09:55.607Z DEBUG topgrade::runner   > Step "pkgin"
 2022-02-28T14:09:55.607Z DEBUG topgrade::utils    > Path "/home/arch/.config/emacs" doesn't exist
 2022-02-28T14:09:55.607Z DEBUG topgrade::utils    > Path "/home/arch/.emacs.d" exists
 2022-02-28T14:09:55.607Z DEBUG topgrade::utils    > Path "/home/arch/.emacs.d/bin/doom" doesn't exist
 2022-02-28T14:09:55.607Z DEBUG topgrade::steps::git > Checking if /home/arch/.emacs.d is a git repository
 2022-02-28T14:09:55.608Z TRACE topgrade::executor   > Output of "/usr/bin/git" "rev-parse" "--show-toplevel": Output { status: ExitStatus(unix_wait_status(32768)), stdout: "", stderr: "fatal: not a git repository (or any parent up to mount point /)\nStopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).\n" }
 2022-02-28T14:09:55.608Z DEBUG topgrade::steps::git > /home/arch/.doom.d does not exists
 2022-02-28T14:09:55.608Z DEBUG topgrade::steps::git > /home/arch/.vim does not exists
 2022-02-28T14:09:55.608Z DEBUG topgrade::steps::git > Checking if /home/arch/.config/nvim is a git repository
 2022-02-28T14:09:55.609Z TRACE topgrade::executor   > Output of "/usr/bin/git" "rev-parse" "--show-toplevel": Output { status: ExitStatus(unix_wait_status(32768)), stdout: "", stderr: "fatal: not a git repository (or any parent up to mount point /)\nStopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).\n" }
 2022-02-28T14:09:55.609Z DEBUG topgrade::steps::git > /home/arch/.ideavimrc does not exists
 2022-02-28T14:09:55.609Z DEBUG topgrade::steps::git > /home/arch/.intellimacs does not exists
 2022-02-28T14:09:55.609Z DEBUG topgrade::steps::git > /home/arch/.zshrc is a file. Checking /home/arch
 2022-02-28T14:09:55.609Z DEBUG topgrade::steps::git > Checking if /home/arch is a git repository
 2022-02-28T14:09:55.610Z TRACE topgrade::executor   > Output of "/usr/bin/git" "rev-parse" "--show-toplevel": Output { status: ExitStatus(unix_wait_status(32768)), stdout: "", stderr: "fatal: not a git repository (or any parent up to mount point /)\nStopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).\n" }
 2022-02-28T14:09:55.610Z DEBUG topgrade::steps::git > /home/arch/.tmux does not exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::steps::git > Checking if /home/arch/.config/fish is a git repository
 2022-02-28T14:09:55.610Z TRACE topgrade::executor   > Output of "/usr/bin/git" "rev-parse" "--show-toplevel": Output { status: ExitStatus(unix_wait_status(32768)), stdout: "", stderr: "fatal: not a git repository (or any parent up to mount point /)\nStopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).\n" }
 2022-02-28T14:09:55.610Z DEBUG topgrade::steps::git > /home/arch/.config/openbox does not exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::steps::git > /home/arch/.config/bspwm does not exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::steps::git > /home/arch/.config/i3 does not exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::steps::git > /home/arch/.config/sway does not exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::runner     > Step "Git repositories"
 2022-02-28T14:09:55.610Z DEBUG topgrade::runner     > Step "zr"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Detected "/usr/bin/zsh" as "zsh"
 2022-02-28T14:09:55.610Z DEBUG topgrade::runner     > Step "antibody"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Detected "/usr/bin/zsh" as "zsh"
 2022-02-28T14:09:55.610Z DEBUG topgrade::runner     > Step "antigen"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Detected "/usr/bin/zsh" as "zsh"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Path "/home/arch/.zshrc" exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::runner     > Step "zplug"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Detected "/usr/bin/zsh" as "zsh"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Path "/home/arch/.zshrc" exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::runner     > Step "zinit"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Detected "/usr/bin/zsh" as "zsh"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Path "/home/arch/.zshrc" exists
 2022-02-28T14:09:55.610Z DEBUG topgrade::runner     > Step "zim"
 2022-02-28T14:09:55.610Z DEBUG topgrade::utils      > Detected "/usr/bin/zsh" as "zsh"
 2022-02-28T14:09:55.613Z TRACE topgrade::executor   > Output of "zsh" "-c" "[[ -n ${ZIM_HOME} ]] && print -n ${ZIM_HOME}": Output { status: ExitStatus(unix_wait_status(256)), stdout: "", stderr: "" }
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "oh-my-zsh"
 2022-02-28T14:09:55.613Z DEBUG topgrade::utils      > Detected "/usr/bin/zsh" as "zsh"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "fisher"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "bash-it"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "oh-my-fish"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "fish-plug"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "tmux"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "TLDR"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "pearl"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "Gnome Shell Extensions"
 2022-02-28T14:09:55.613Z DEBUG topgrade::utils      > Detected "/usr/bin/gdbus" as "gdbus"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "SDKMAN!"
 2022-02-28T14:09:55.613Z DEBUG topgrade::utils      > Detected "/usr/bin/bash" as "bash"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "apm"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "fossil"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "rustup"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step ".NET"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "choosenim"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "cargo"
 2022-02-28T14:09:55.613Z DEBUG topgrade::utils      > Path "/home/arch/.cargo" exists
 2022-02-28T14:09:55.613Z DEBUG topgrade::utils      > Detected "/usr/bin/cargo" as "cargo"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "Flutter"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "Go"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "Emacs"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "opam"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "vcpkg"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "pipx"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "conda"
 2022-02-28T14:09:55.613Z DEBUG topgrade::runner     > Step "pip3"
 2022-02-28T14:09:55.613Z DEBUG topgrade::utils      > Detected "/home/arch/.local/bin/pip3" as "pip3"

―― 19:39:55 - pip3 ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 2022-02-28T14:09:55.613Z DEBUG topgrade::executor   > Running "/home/arch/.local/bin/pip3" "install" "--upgrade" "--user" "pip"
Requirement already satisfied: pip in ./.local/lib/python3.10/site-packages (22.0.3)
 2022-02-28T14:09:57.172Z DEBUG topgrade::runner     > Step "stack"
 2022-02-28T14:09:57.172Z DEBUG topgrade::runner     > Step "tlmgr"
 2022-02-28T14:09:57.172Z DEBUG topgrade::runner     > Step "myrepos"
 2022-02-28T14:09:57.172Z DEBUG topgrade::runner     > Step "chezmoi"
 2022-02-28T14:09:57.172Z DEBUG topgrade::runner     > Step "jetpack"
 2022-02-28T14:09:57.172Z DEBUG topgrade::runner     > Step "vim"
 2022-02-28T14:09:57.173Z DEBUG topgrade::utils      > Detected "/usr/bin/vim" as "vim"
 2022-02-28T14:09:57.174Z TRACE topgrade::executor   > Output of "/usr/bin/vim" "--version": Output { status: ExitStatus(unix_wait_status(0)), stdout: "VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Feb 24 2022 19:44:18)\nIncluded patches: 1-4464\nCompiled by Arch Linux\nHuge version without GUI.  Features included (+) or not (-):\n+acl               +file_in_path      +mouse_urxvt       -tag_any_white\n+arabic            +find_in_path      +mouse_xterm       +tcl/dyn\n+autocmd           +float             +multi_byte        +termguicolors\n+autochdir         +folding           +multi_lang        +terminal\n-autoservername    -footer            -mzscheme          +terminfo\n-balloon_eval      +fork()            +netbeans_intg     +termresponse\n+balloon_eval_term +gettext           +num64             +textobjects\n-browse            -hangul_input      +packages          +textprop\n++builtin_terms    +iconv             +path_extra        +timers\n+byte_offset       +insert_expand     +perl/dyn          +title\n+channel           +ipv6              +persistent_undo   -toolbar\n+cindent           +job               +popupwin          +user_commands\n-clientserver      +jumplist          +postscript        +vartabs\n-clipboard         +keymap            +printer           +vertsplit\n+cmdline_compl     +lambda            +profile           +vim9script\n+cmdline_hist      +langmap           -python            +viminfo\n+cmdline_info      +libcall           +python3/dyn       +virtualedit\n+comments          +linebreak         +quickfix          +visual\n+conceal           +lispindent        +reltime           +visualextra\n+cryptv            +listcmds          +rightleft         +vreplace\n+cscope            +localmap          +ruby/dyn          +wildignore\n+cursorbind        +lua/dyn           +scrollbind        +wildmenu\n+cursorshape       +menu              +signs             +windows\n+dialog_con        +mksession         +smartindent       +writebackup\n+diff              +modify_fname      -sodium            -X11\n+digraphs          +mouse             -sound             -xfontset\n-dnd               -mouseshape        +spell             -xim\n-ebcdic            +mouse_dec         +startuptime       -xpm\n+emacs_tags        +mouse_gpm         +statusline        -xsmp\n+eval              -mouse_jsbterm     -sun_workshop      -xterm_clipboard\n+ex_extra          +mouse_netterm     +syntax            -xterm_save\n+extra_search      +mouse_sgr         +tag_binary        \n-farsi             -mouse_sysmouse    -tag_old_static    \n   system vimrc file: \"/etc/vimrc\"\n     user vimrc file: \"$HOME/.vimrc\"\n 2nd user vimrc file: \"~/.vim/vimrc\"\n      user exrc file: \"$HOME/.exrc\"\n       defaults file: \"$VIMRUNTIME/defaults.vim\"\n  fall-back for $VIM: \"/usr/share/vim\"\nCompilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -march=x86-64-v3 -O3 -pipe -fno-plt -fexceptions -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 \nLinking: gcc -L. -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector-strong -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.34/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -L/usr/local/lib -o vim -lm -ltinfo -lelf -lacl -lattr -lgpm -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.34/core_perl/CORE -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fstack-protector-strong -L/usr/local/lib -L/usr/lib/perl5/5.34/core_perl/CORE -lperl -lpthread -ldl -lm -lcrypt -lutil -lc -L/usr/lib -ltclstub8.6 -ldl -lz -lpthread -lm \n", stderr: "" }
 2022-02-28T14:09:57.175Z DEBUG topgrade::runner     > Step "Neovim"
 2022-02-28T14:09:57.175Z DEBUG topgrade::runner     > Step "voom"
 2022-02-28T14:09:57.175Z DEBUG topgrade::runner     > Step "npm"
 2022-02-28T14:09:57.175Z DEBUG topgrade::utils      > Detected "/usr/bin/npm" as "npm"
 2022-02-28T14:09:57.441Z TRACE topgrade::executor   > Output of "/usr/bin/npm" "root" "-g": Output { status: ExitStatus(unix_wait_status(0)), stdout: "/usr/lib/node_modules\n", stderr: "" }
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "pnpm"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "deno"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "composer"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "krew"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "gem"
 2022-02-28T14:09:57.441Z DEBUG topgrade::utils      > Detected "/usr/bin/gem" as "gem"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "haxelib"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "sheldon"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "rtcl"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "bin"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "gcloud"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "micro"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "raco"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "spicetify"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "toolbx"
 2022-02-28T14:09:57.441Z DEBUG topgrade::runner     > Step "Flatpak"
 2022-02-28T14:09:57.441Z DEBUG topgrade::utils      > Detected "/usr/bin/flatpak" as "flatpak"

―― 19:39:57 - Flatpak User Packages ――――――――――――――――――――――――――――――――――――――――――――
 2022-02-28T14:09:57.441Z DEBUG topgrade::executor   > Running "/usr/bin/flatpak" "update" "--user" "-y"
Looking for updates…
Nothing to do.

―― 19:39:57 - Flatpak System Packages ――――――――――――――――――――――――――――――――――――――――――
 2022-02-28T14:09:57.464Z DEBUG topgrade::executor   > Running "/usr/bin/flatpak" "update" "--system" "-y"
Looking for updates…
Nothing to do.
 2022-02-28T14:09:58.336Z DEBUG topgrade::runner     > Step "snap"
 2022-02-28T14:09:58.336Z DEBUG topgrade::runner     > Step "pacstall"
 2022-02-28T14:09:58.336Z DEBUG topgrade::runner     > Step "Pacman files DB"

―― 19:39:58 - Pacman files DB ――――――――――――――――――――――――――――――――――――――――――――――――――
 2022-02-28T14:09:58.336Z DEBUG topgrade::executor   > Running "/usr/bin/zsh" "-c" "sudo pacman -Fy"
:: Synchronizing package databases...
 core-x86-64-v3 is up to date
 extra-x86-64-v3 is up to date
 community-x86-64-v3 is up to date
 core is up to date
 extra is up to date
 community is up to date
 2022-02-28T14:10:00.939Z DEBUG topgrade::runner     > Step "pihole"
 2022-02-28T14:10:00.939Z DEBUG topgrade::runner     > Step "Firmware upgrades"
 2022-02-28T14:10:00.939Z DEBUG topgrade::utils      > Detected "/usr/bin/fwupdmgr" as "fwupdmgr"
 2022-02-28T14:10:00.939Z TRACE topgrade::executor   > Output of "uname" "-r": Output { status: ExitStatus(unix_wait_status(0)), stdout: "5.15.25-1.1-lts\n", stderr: "" }
 2022-02-28T14:10:00.939Z DEBUG topgrade::steps::os::linux > Uname output: 5.15.25-1.1-lts

―― 19:40:00 - Firmware upgrades ――――――――――――――――――――――――――――――――――――――――――――――――
 2022-02-28T14:10:00.939Z DEBUG topgrade::executor         > Running "/usr/bin/fwupdmgr" "refresh"
Updating lvfs
Downloading…             [***************************************]
Successfully downloaded new metadata: 0 local devices supported
 2022-02-28T14:10:01.466Z DEBUG topgrade::executor         > Running "/usr/bin/fwupdmgr" "get-updates"
Devices with no available firmware updates: 
 • UEFI dbx
No updatable devices
 2022-02-28T14:10:01.532Z DEBUG topgrade::runner           > Step "Restarts"

―― 19:40:01 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
System update: OK
config-update: OK
Brew: SKIPPED: Cannot find "brew" in PATH
nix: SKIPPED: Cannot find "nix" in PATH
home-manager: SKIPPED: Cannot find "home-manager" in PATH
asdf: SKIPPED: Cannot find "asdf" in PATH
pkgin: SKIPPED: Cannot find "pkgin" in PATH
Git repositories: SKIPPED: No repositories to pull
zr: SKIPPED: Cannot find "zr" in PATH
antibody: SKIPPED: Cannot find "antibody" in PATH
antigen: SKIPPED: Path "/home/arch/antigen.zsh" doesn't exist
zplug: SKIPPED: Path "/home/arch/.zplug" doesn't exist
zinit: SKIPPED: Path "/home/arch/.zinit" doesn't exist
zim: SKIPPED: Path "/home/arch/.zim" doesn't exist
oh-my-zsh: SKIPPED: Path "/home/arch/.oh-my-zsh" doesn't exist
fisher: SKIPPED: Cannot find "fish" in PATH
bash-it: SKIPPED: Path "/home/arch/.bash_it" doesn't exist
oh-my-fish: SKIPPED: Cannot find "fish" in PATH
fish-plug: SKIPPED: Cannot find "fish" in PATH
tmux: SKIPPED: Path "/home/arch/.tmux/plugins/tpm/bin/update_plugins" doesn't exist
TLDR: SKIPPED: Cannot find "tldr" in PATH
pearl: SKIPPED: Cannot find "pearl" in PATH
Gnome Shell Extensions: SKIPPED: Desktop doest not appear to be gnome
SDKMAN!: SKIPPED: Path "/home/arch/.sdkman/bin/sdkman-init.sh" doesn't exist
apm: SKIPPED: Cannot find "apm" in PATH
fossil: SKIPPED: Cannot find "fossil" in PATH
rustup: SKIPPED: Cannot find "rustup" in PATH
.NET: SKIPPED: Cannot find "dotnet" in PATH
choosenim: SKIPPED: Cannot find "choosenim" in PATH
cargo: SKIPPED: Path "/home/arch/.cargo/.crates.toml" doesn't exist
Flutter: SKIPPED: Cannot find "flutter" in PATH
Go: SKIPPED: Cannot find "go" in PATH
Emacs: SKIPPED: Cannot find "emacs" in PATH
opam: SKIPPED: Cannot find "opam" in PATH
vcpkg: SKIPPED: Cannot find "vcpkg" in PATH
pipx: SKIPPED: Cannot find "pipx" in PATH
conda: SKIPPED: Cannot find "conda" in PATH
pip3: OK
stack: SKIPPED: Cannot find "stack" in PATH
tlmgr: SKIPPED: tlmgr must be explicity enabled in the configuration to run in Linux
myrepos: SKIPPED: Cannot find "mr" in PATH
chezmoi: SKIPPED: Cannot find "chezmoi" in PATH
jetpack: SKIPPED: Cannot find "jetpack" in PATH
vim: SKIPPED: Path "/home/arch/.vim/vimrc" doesn't exist
Neovim: SKIPPED: Cannot find "nvim" in PATH
voom: SKIPPED: Cannot find "voom" in PATH
npm: SKIPPED: NPM root is owned by another user which is not the current user. Set use_sudo = true under the NPM section in your configuration to run NPM as sudo
pnpm: SKIPPED: Cannot find "pnpm" in PATH
deno: SKIPPED: Cannot find "deno" in PATH
composer: SKIPPED: Cannot find "composer" in PATH
krew: SKIPPED: Cannot find "kubectl-krew" in PATH
gem: SKIPPED: Path "/home/arch/.gem" doesn't exist
haxelib: SKIPPED: Cannot find "haxelib" in PATH
sheldon: SKIPPED: Cannot find "sheldon" in PATH
rtcl: SKIPPED: Cannot find "rupdate" in PATH
bin: SKIPPED: Cannot find "bin" in PATH
gcloud: SKIPPED: Cannot find "gcloud" in PATH
micro: SKIPPED: Cannot find "micro" in PATH
raco: SKIPPED: Cannot find "raco" in PATH
spicetify: SKIPPED: Cannot find "spicetify" in PATH
toolbx: SKIPPED: Cannot find "toolbox" in PATH
Flatpak: OK
snap: SKIPPED: Cannot find "snap" in PATH
pacstall: SKIPPED: Cannot find "pacstall" in PATH
Pacman files DB: OK
pihole: SKIPPED: Cannot find "pihole" in PATH
Firmware upgrades: OK
Restarts: SKIPPED: Cannot find "needrestart" in PATH
 2022-02-28T14:10:01.535Z DEBUG topgrade::terminal         > Desktop notification: Topgrade finished successfully

</pre></details>

**Comment by r-darwish** [comment]
Why is this a bad thing? Dry run doesn't mean Topgrade won't run any command, it means it won't update anything.

**Comment by Samisafool** [comment]
The news command still does something. While that might not be updating, I still think dry-run should not run anu commands, as dry run means it won't do *anything* except display what might have been done.

**Comment by MCOfficer** [comment]
> The news command still does something. While that might not be updating, I still think dry-run should not run any commands, as dry run means it won't do _anything_ except display what might have been done.

There will always be some commands that are executed, though. For example, `pacman -Sy` and `scoop update` are needed to even display the available updates.

**Comment by Samisafool** [comment]
> There will always be some commands that are executed, though. For example, pacman -Sy and scoop update are needed to even display the available updates.

Wait, what? Since when does topgrade display those available updates?

**Comment by MCOfficer** [comment]
> > There will always be some commands that are executed, though. For example, pacman -Sy and scoop update are needed to even display the available updates.
> 
> Wait, what? Since when does topgrade display those available updates?

Sorry, I'd just woken up and confused a bunch of things. [Selfie at the time of posting](https://media.tenor.com/images/b21f659219b6ff3467eb2526074b227a/tenor.gif)...

**Comment by Samisafool** [comment]
Why was this closed? I would appreciate atleast an option in the config for it...
